cmd/auth-provider-gcp/provider/BUILD

```diff
@@ -15,9 +15,7 @@ go_library(
         "//pkg/apis/credentialprovider:go_default_library",
```

Comment:
I think we can remove the entire pkg/apis/credentialprovider folder now that we're supporting 1.20 - this package is now in staging/ at `k8s.io/kubelet/pkg/apis/credentialprovider` as a part of [this PR](https://github.com/kubernetes/kubernetes/pull/94196). This is the only package in the repo that depends upon `pkg/apis/credentialprovider`, I believe, so the entire pkg/apis/credentialprovider directory should be able to be safely removed afterward.

Reply:
Thank you for the reminder. Updated to remove this.